crux-core/src/crux/api/alpha/CruxNode.java

```diff
@@ -93,17 +61,42 @@ private Document document(Object contentHash) {
     }
 
     /**
-     * Returns the transaction history of an entity, in reverse chronological order. Includes corrections, but does not include the actual documents
+     * Returns the transaction history of an entity, in reverse chronological order. Includes corrections, but does not include the actual documents.
      * @param id Id of the entity to get the history for
-     * @return Iterable set of Documents containing transaction information
+     * @return Iterable set of EntityTx's containing transaction information
+     * @see EntityTx
      */
-    public Iterable<Document> history(CruxId id) {
+    public Iterable<EntityTx> history(CruxId id) {
         List<Map<Keyword,Object>> history = node.history(id.toEdn());
         return history.stream()
-            .map(this::document)
+            .map(entity -> EntityTx.entityTx(entity))
             .collect(Collectors.toList());
     }
 
+    /**
```

Comment:
@danmason - can you omit these two (`hasSubmittedTx*`) for the time being? not convinced they're going to remain supported

Reply:
Yeah, can do - remove, or make private?